src/NHibernate/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -90,7 +90,7 @@ public void ExecuteWorkInIsolation(ISessionImplementor session, IIsolatedWork wo
 			}
 		}
 
-		public class DistributedTransactionContext : ITransactionContext, IEnlistmentNotification
+		public class DistributedTransactionContext : ITransactionContext, IEnlistmentNotification, IPromotableSinglePhaseNotification
```

Comment:
You are correct about `IPromotableSinglePhaseNotification` not actually being needed.  When I added the cleanup `CheckEnlistedTransactionsWereDisposed` in `TestCase.cs` is when most of the errors went away.

I'll revert the changes to `DistributedTransactionContext`.